rio/src/RIO/Prelude/RIO.hs

```diff
@@ -35,3 +52,104 @@ instance MonadUnliftIO (RIO env) where
 instance PrimMonad (RIO env) where
     type PrimState (RIO env) = PrimState IO
     primitive = RIO . ReaderT . const . primitive
+
+-- | Abstraction over how to read from and write to a mutable reference
+--
+-- @since 0.1.3.0
+data SomeRef a
+  = SomeRef !(IO a) !(a -> IO ())
+
+-- | Read from a SomeRef
+--
+-- @since 0.1.3.0
+readSomeRef :: SomeRef a -> IO a
```

Comment:
Only noticing this now: for consistency with the rest of rio, this should be lifted to `MonadIO`.

Reply:
Makes sense, I'm wondering though if the definition of SomeRef is still oke? Wouldn't it be better if the SomeRef knew something about MonadIO instead of being (SomeRef !(IO a) !(a -> IO a))? As far as i know, we cannot add a typeclass constraint on the creation of the datatype but i found this post that says its possible with GADTs (https://wiki.haskell.org/Data_declaration_with_constraint)